addons/udes_stock/models/res_users.py

```diff
@@ -37,3 +67,59 @@ def get_user_warehouse(self, aux_domain=None):
                 )
             )
         return warehouse
+
+    def assign_picking_to_users(self, picking):
+        """
+        Assign a picking to user(s) in self
+        """
+        picking.ensure_one()
+        picking_type = picking.picking_type_id
+
+        # Find the users to assign
+        users_to_assign = self.filtered(lambda u: u.u_picking_id != picking)
+        if users_to_assign:
+            if picking.state in NON_ASSIGNABLE_STATES:
+                _logger.warning(
+                    "Cannot assign users %s to picking %s as it is in state %s"
+                    % (self.mapped("name"), picking.name, picking.state)
+                )
+            elif len(self) > 1 and not picking_type.can_handle_multiple_users():
+                _logger.warning(
+                    "Cannot assign users %s to picking %s as picking type %s only allows one user"
+                    % (self.mapped("name"), picking.name, picking_type.name)
+                )
+            else:
+                # Un assign the current users from their current (different) picking
+                users_to_assign.un_assign_pickings_from_users(new_picking=picking)
+                # Kick out any users from the current picking if needed
+                # Do not kick the current users in self out as they are already unassigned
+                # from their own pickings.
+                # NOTE: We could check here if there are any users in the picking, and kick them
+                # out if needed, but instead we just extend unassign_users, as this
+                # is needed when closing batches or if we extend the behaviour of multiple users
+                # working on a picking.
+                picking.un_assign_users(skip_users=users_to_assign)
+                users_to_assign._assign_picking_to_users(picking)
+
+    def _assign_picking_to_users(self, picking):
+        """Placing into a method in order to be able to inherit and change behaviour if needed.
+        If the picking has never been started, update the start time."""
+        now = PreciseDatetime.now()
+        self.sudo().write(
```

Comment:
Q: Are the sudo calls required for 14 son the user model?

Reply:
What does it mean "son", i think sudos are needed when we want to be sure that user have the right to run the code even if he hasnt the right to access in this case to write. And that is even for udes14 i think, at least saw some scenarios in udes14 odoo core modules where was used

Reply:
Probably "unassign" is not an "official" word, so it isn't in the dictionary, but in general "un" and the wod it negates aren't separated.  And "unassign" is already used else in the Odoo codebase, we should be consistent with that.

Reply:
sorry typo son -> on.

Ah no we need it because it can assign multiple users info. I was just wondering if you can write on your own record without permission issues. But perhaps thinking about it that is probably not the case, as they could give themselvees to much power.

Reply:
Making the changes of the methods again :)